File: PathSum/PathSum.java
/*
 //Definition for a binary tree node.
 public class TreeNode {
     int val;
     TreeNode left;
     TreeNode right;
     TreeNode() {}
     TreeNode(int val) { this.val = val; }
     TreeNode(int val, TreeNode left, TreeNode right) {
         this.val = val;
         this.left = left;
         this.right = right;
     }
}

*/
class Solution {
    public boolean hasPathSum(TreeNode root, int targetSum) {
        boolean isFound = false;
        int sum = targetSum;

        if(root == null ){
            return false;
        }
        if(root.left == null && root.right == null){ //check leaf node
            if(sum - root.val == 0){
                return true;
            }
            return false;
        }
        sum -= root.val;
        if(root.left != null){
           isFound = hasPathSum(root.left, sum);
        }
        if(root.right != null && isFound == false){
           isFound =  hasPathSum(root.right, sum);
        }
        return isFound;
    }
}

File: PathSum/PathSum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool hasPathSum(TreeNode* root, int targetSum) {
        bool isFound = false;
        int sum = targetSum;
        if(root == NULL ){
            return false;
        }
        if(root->left == NULL && root->right == NULL){ //check leaf node
            if(sum - root->val == 0){
                return true;
            }
            return false;
        }
        sum -= root->val;
        if(root->left != NULL){
           isFound = hasPathSum(root->left, sum);
        }
        if(root->right != NULL && isFound == false){
           isFound =  hasPathSum(root->right, sum);
        }
        return isFound;
    }
};